crane/data.py

```diff
@@ -38,18 +42,23 @@ def load_from_file(path):
         repo_data = json.load(json_file)
 
     # for now, we only support version 1 of the metadata schema
-    if repo_data['version'] != 1:
+    if repo_data['version'] not in (1, 2):
         raise ValueError('metadata version %d not supported' % repo_data['version'])
 
     repo_id = repo_data['repo-registry-id']
-    image_ids = [image['id'] for image in repo_data['images']]
     url_path = urlparse.urlparse(repo_data['url']).path
-    repo_tuple = Repo(repo_data['url'],
-                      json.dumps(repo_data['images']),
-                      json.dumps(repo_data['tags']),
-                      url_path, repo_data.get('protected', False))
 
-    return repo_id, repo_tuple, image_ids
+    if repo_data['version'] == 1:
+        image_ids = [image['id'] for image in repo_data['images']]
+        repo_tuple = Repo(repo_data['url'],
+                          json.dumps(repo_data['images']),
+                          json.dumps(repo_data['tags']),
+                          url_path, repo_data.get('protected', False))
+        return repo_id, repo_tuple, image_ids
+    elif repo_data['version'] == 2:
+        repo_tuple = Repo_v2(repo_data['url'],
+                             url_path, repo_data.get('protected', False))
+        return repo_id, repo_tuple, None
```

Comment:
Since v2 returns None for images, please adjust the docblock for this function to mention that this will return None for v2.